src/App.tsx

```diff
@@ -15,35 +16,98 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  BY_LENGTH = 'length',
+  ALPHABETICALLY = 'alphabetically',
+  DEFAULT = '',
+}
+
 export const App: React.FC = () => {
+  const [sortField, setSortField] = useState<SortType>(SortType.DEFAULT);
+  const [sortOrder, setSortOrder] = useState<boolean>(true);
+
+  function prepareGoods(goods: string[], sortBy: string, order = true) {
+    const preparedGoods = [...goods];
+
+    switch (sortBy) {
+      case SortType.BY_LENGTH: {
+        preparedGoods.sort((goodA, goodB) => goodA.length - goodB.length);
+        break;
+      }
+
+      case SortType.ALPHABETICALLY: {
+        preparedGoods.sort((goodA, goodB) => goodA.localeCompare(goodB));
+        break;
+      }
+
+      default:
+        break;
+    }
+
+    return order ? preparedGoods : preparedGoods.reverse();
+  }
+
+  function resetHandler(): void {
+    setSortField(SortType.DEFAULT);
+    setSortOrder(true);
+  }
+
+  const visibleGoods = prepareGoods(goodsFromServer, sortField, sortOrder);
+
   return (
     <div className="section content">
       <div className="buttons">
-        <button type="button" className="button is-info is-light">
-          Sort alphabetically
+        <button
+          type="button"
+          className={classNames('button', 'is-info', {
+            'is-light': sortField !== SortType.ALPHABETICALLY,
```

Comment:
To comply with the [UX] checklist, ensure that the active button is marked clearly. The current approach is correct, but it's important to verify that the 'is-light' class effectively indicates the inactive state to the user.